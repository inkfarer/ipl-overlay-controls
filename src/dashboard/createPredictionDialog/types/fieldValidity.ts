export interface FieldValidity{
    [key: string]: boolean
}