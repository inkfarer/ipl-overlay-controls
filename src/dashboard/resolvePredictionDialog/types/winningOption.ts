export interface WinningOption{
    validOption: boolean;
    optionIndex: number;
    optionTitle: string;
}