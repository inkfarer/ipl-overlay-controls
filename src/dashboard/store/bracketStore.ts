import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
