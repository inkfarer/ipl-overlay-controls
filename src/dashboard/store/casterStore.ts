import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const useCasterStore = defineStore('casters', {
    state: () => ({
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    } as CasterStore),
    actions: {
        updateCaster({ id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        removeUncommittedCaster(id: string): void {
            delete this.uncommittedCasters[id];
        },
        addUncommittedCaster({ id, caster }: { id: string, caster?: Caster }): void {
            this.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        addDefaultCaster(): string {
            const id = generateId();
            this.addUncommittedCaster({ id });
            return id;
        },
        async saveUncommittedCaster({ id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            this.removeUncommittedCaster(id);
            return newId;
        },
        async loadCastersFromVc(): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    this.addUncommittedCaster({ id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        async removeCaster(id: string): Promise<void> {
            return nodecg.sendMessage('removeCaster', { id });
        }
    }
});
