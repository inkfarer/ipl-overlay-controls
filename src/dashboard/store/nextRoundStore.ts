import { NextRound } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { BeginNextMatchRequest } from 'types/messages/activeRound';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const nextRoundStore = createStore<NextRoundStore>({
    state: {
        nextRound: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        },
        setShowOnStream(store, newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        }
    },
    actions: {
        beginNextMatch(store, request: BeginNextMatchRequest): void {
            nodecg.sendMessage('beginNextMatch', request);
        },
        setNextRound(store, {
            teamAId,
            teamBId,
            roundId
        }: { teamAId: string, teamBId: string, roundId: string }): void {
            nodecg.sendMessage('setNextRound', { teamAId, teamBId, roundId } as SetNextRoundRequest);
        },
    }
});

export const nextRoundStoreKey: InjectionKey<Store<NextRoundStore>> = Symbol();

export function useNextRoundStore(): Store<NextRoundStore> {
    return useStore(nextRoundStoreKey);
}
