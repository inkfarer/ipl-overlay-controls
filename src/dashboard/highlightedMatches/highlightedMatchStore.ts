import { NodeCGBrowser } from 'nodecg/browser';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import {
    HighlightedMatches,
    TournamentData
} from 'schemas';
import { SelectOptions } from '../types/select';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';

const highlightedMatches = nodecg.Replicant<HighlightedMatches>('highlightedMatches');
const tournamentData = nodecg.Replicant<TournamentData>('tournamentData');

export const highlightedMatchReps = [ highlightedMatches, tournamentData ];

export interface HighlightedMatchStore {
    highlightedMatches: HighlightedMatches;
    tournamentData: TournamentData;
}

export const highlightedMatchStore = createStore<HighlightedMatchStore>({
    state: {
        highlightedMatches: null,
        tournamentData: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async getHighlightedMatches(store, { options }: { options: SelectOptions }): Promise<void> {
            const getAllMatches = options.some(option => option.value === 'all');

            if (getAllMatches) {
                return nodecg.sendMessage('getHighlightedMatches', { getAllMatches });
            } else {
                const values = options.map(option => option.value);

                switch (store.state.tournamentData.meta.source) {
                    case TournamentDataSource.BATTLEFY:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            stages: values
                        });
                    case TournamentDataSource.SMASHGG:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            streamIDs: values.map(value => parseInt(value))
                        });
                    default:
                        throw new Error(`Cannot import data from source '${store.state.tournamentData.meta.source}'`);
                }
            }
        },
        async setNextMatch(store, { teamAId, teamBId }: { teamAId: string, teamBId: string }): Promise<void> {
            return nodecg.sendMessage('setNextRound', { teamAId, teamBId });
        }
    }
});

export const highlightedMatchStoreKey: InjectionKey<Store<HighlightedMatchStore>> = Symbol();

export function useHighlightedMatchStore(): Store<HighlightedMatchStore> {
    return useStore(highlightedMatchStoreKey);
}
