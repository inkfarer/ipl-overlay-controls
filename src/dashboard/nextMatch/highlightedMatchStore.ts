import { NodeCGBrowser } from 'nodecg/browser';
import { HighlightedMatches, TournamentData } from 'schemas';
import { SelectOptions } from '../types/select';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { defineStore } from 'pinia';

const highlightedMatches = nodecg.Replicant<HighlightedMatches>('highlightedMatches');
const tournamentData = nodecg.Replicant<TournamentData>('tournamentData');

export const highlightedMatchReps = [ highlightedMatches, tournamentData ];

export interface HighlightedMatchStore {
    highlightedMatches: HighlightedMatches;
    tournamentData: TournamentData;
}

export const useHighlightedMatchStore = defineStore('highlightedMatches', {
    state: () => ({
        highlightedMatches: null,
        tournamentData: null
    } as HighlightedMatchStore),
    actions: {
        async getHighlightedMatches({ options }: { options: SelectOptions }): Promise<void> {
            const getAllMatches = options.some(option => option.value === 'all');

            if (getAllMatches) {
                return nodecg.sendMessage('getHighlightedMatches', { getAllMatches });
            } else {
                const values = options.map(option => option.value);

                switch (this.tournamentData.meta.source) {
                    case TournamentDataSource.BATTLEFY:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            stages: values
                        });
                    case TournamentDataSource.SMASHGG:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            streamIDs: values.map(value => parseInt(value))
                        });
                    default:
                        throw new Error(`Cannot import data from source '${this.tournamentData.meta.source}'`);
                }
            }
        },
        async setNextMatch(data: SetNextRoundRequest): Promise<void> {
            return nodecg.sendMessage('setNextRound', data);
        }
    }
});
