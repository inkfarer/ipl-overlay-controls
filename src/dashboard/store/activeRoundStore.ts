import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
