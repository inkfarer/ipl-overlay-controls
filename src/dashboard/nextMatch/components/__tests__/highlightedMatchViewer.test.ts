import HighlightedMatchViewer from '../highlightedMatchViewer.vue';
import { createStore } from 'vuex';
import { HighlightedMatchStore, highlightedMatchStoreKey } from '../../highlightedMatchStore';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import { config, mount } from '@vue/test-utils';
import { Team } from 'types/team';
import { NextRoundStore, nextRoundStoreKey } from '../../../store/nextRoundStore';

describe('HighlightedMatchViewer', () => {
    config.global.stubs = {
        IplSelect: true,
        IplDataRow: true,
        IplButton: true,
        RoundSelect: true,
        FontAwesomeIcon: true
    };

    const mockSetNextMatch = jest.fn();
    const mockTeam: Team = { id: '1234', name: 'mock team', showLogo: false, players: []};

    function createNextRoundStore() {
        return createStore<NextRoundStore>({
            state: {
                nextRound: {
                    teamA: { id: '123123', name: 'cool team A', showLogo: true, players: []},
                    teamB: { id: '345345', name: 'cool team B', showLogo: false, players: []},
                    round: { id: '0387', name: 'dope round' },
                    showOnStream: true,
                    games: []
                }
            }
        });
    }

    function createHighlightedMatchStore() {
        return createStore<HighlightedMatchStore>({
            state: {
                tournamentData: {
                    meta: {
                        id: 'tournament123',
                        source: TournamentDataSource.UNKNOWN
                    },
                    teams: []
                },
                highlightedMatches: null
            },
            actions: {
                setNextMatch: mockSetNextMatch
            }
        });
    }

    it('matches snapshot when no matches are loaded', () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot', () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'cool match', id: '1234' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'cooler match', id: '567' }, teamA: mockTeam, teamB: mockTeam }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('selects first match when highlighted matches are updated', async () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'cool match', id: '1234' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'cooler match', id: '567' }, teamA: mockTeam, teamB: mockTeam }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        expect(wrapper.getComponent('[data-test="match-selector"]').attributes().modelvalue).toEqual('1234');
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'new match', id: '678' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'new match 2', id: '3089754' }, teamA: mockTeam, teamB: mockTeam }
        ];
        await wrapper.vm.$nextTick();
        expect(wrapper.getComponent('[data-test="match-selector"]').attributes().modelvalue).toEqual('678');
    });

    it('selects match with same teams as next round if found', async () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'cool match', id: '1234' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'cooler match', id: '567' }, teamA: mockTeam, teamB: mockTeam }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        expect(wrapper.getComponent('[data-test="match-selector"]').attributes().modelvalue).toEqual('1234');
        const mockSelectedTeamA = { ...mockTeam, id: '123123' };
        const mockSelectedTeamB = { ...mockTeam, id: '345345' };
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'new match', id: '678' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'new match 2', id: '3089754' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'coolest match', id: '7396' }, teamA: mockSelectedTeamA, teamB: mockSelectedTeamB }
        ];
        await wrapper.vm.$nextTick();
        expect(wrapper.getComponent('[data-test="match-selector"]').attributes().modelvalue).toEqual('7396');
    });

    it('displays team names for selected match', async () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'cool match', id: '1234' }, teamA: mockTeam, teamB: mockTeam },
            {
                meta: { name: 'cooler match', id: '567' },
                teamA: {
                    ...mockTeam,
                    name: 'mock team a'
                },
                teamB: {
                    ...mockTeam,
                    name: 'mock team b'
                }
            }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        wrapper.getComponent('[data-test="match-selector"]').vm.$emit('update:modelValue', '567');
        await wrapper.vm.$nextTick();
        expect(wrapper.getComponent('[data-test="team-a-name-display"]').attributes().value).toEqual('mock team a');
        expect(wrapper.getComponent('[data-test="team-b-name-display"]').attributes().value).toEqual('mock team b');
    });

    it('commits to store on set next match button press', async () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            {
                meta: { name: 'cooler match', id: '567' },
                teamA: {
                    ...mockTeam,
                    id: '1234'
                },
                teamB: {
                    ...mockTeam,
                    id: '5678'
                }
            }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        wrapper.getComponent('[data-test="set-next-match-button"]').vm.$emit('click');

        expect(mockSetNextMatch).toHaveBeenCalledWith(expect.any(Object), {
            teamAId: '1234',
            teamBId: '5678',
            roundId: '0387'
        });
    });

    it('changes update button color when data is changed', async () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            { meta: { name: 'cool match', id: '1234' }, teamA: mockTeam, teamB: mockTeam },
            { meta: { name: 'cooler match', id: '567' }, teamA: mockTeam, teamB: mockTeam }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        wrapper.getComponent('[data-test="match-selector"]').vm.$emit('update:modelValue', '567');
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[data-test="set-next-match-button"]').attributes().color).toEqual('red');
    });

    it('disables set next match button if selected match data is not found', async () => {
        const highlightedMatchStore = createHighlightedMatchStore();
        highlightedMatchStore.state.highlightedMatches = [
            {
                meta: { name: 'cooler match', id: '567' },
                teamA: mockTeam,
                teamB: mockTeam
            }
        ];
        const nextRoundStore = createNextRoundStore();
        const wrapper = mount(HighlightedMatchViewer, {
            global: {
                plugins: [
                    [highlightedMatchStore, highlightedMatchStoreKey],
                    [nextRoundStore, nextRoundStoreKey]
                ]
            }
        });

        wrapper.getComponent('[data-test="match-selector"]').vm.$emit('update:modelValue', 'something');
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[data-test="set-next-match-button"]').attributes().disabled).toEqual('true');
    });
});
