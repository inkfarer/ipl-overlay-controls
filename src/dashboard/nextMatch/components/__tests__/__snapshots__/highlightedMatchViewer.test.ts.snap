// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedMatchViewer matches snapshot 1`] = `
"<div class=\\"ipl-space color-dark\\" data-v-11398c0e=\\"\\">
  <ipl-select-stub modelvalue=\\"1234\\" label=\\"Match\\" options=\\"[object Object],[object Object]\\" disabled=\\"false\\" data-test=\\"match-selector\\"></ipl-select-stub>
  <ipl-data-row-stub label=\\"Team A\\" data-test=\\"team-a-name-display\\" value=\\"mock team\\"></ipl-data-row-stub>
  <ipl-data-row-stub label=\\"Team B\\" data-test=\\"team-b-name-display\\" value=\\"mock team\\"></ipl-data-row-stub>
  <round-select-stub modelvalue=\\"0387\\" class=\\"m-t-4\\"></round-select-stub>
  <ipl-button-stub label=\\"Update\\" color=\\"red\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" href=\\"\\" class=\\"m-t-8\\" data-test=\\"set-next-match-button\\"></ipl-button-stub>
</div>"
`;

exports[`HighlightedMatchViewer matches snapshot when no matches are loaded 1`] = `
"<div class=\\"ipl-message__wrapper layout horizontal center-vertical ipl-message__type-info\\" data-v-13b37b8d=\\"\\">
  <font-awesome-icon-stub icon=\\"info-circle\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"icon\\" data-v-13b37b8d=\\"\\"></font-awesome-icon-stub>
  <div class=\\"message-content\\" data-v-13b37b8d=\\"\\">
    <!----> No matches loaded.
  </div>
</div>"
`;
